### PR TITLE
Reformat dtd metrics job

### PR DESCRIPTION
**Pull request by Carminepo2**
- Added env variable `DEV_FILTER_METRICS`. If given, executes only the metrics that matches the given filter.
- Added env variable `DEV_PRODUCE_JSON`. If given, produces a JSON files containing the metrics output.
- General code reformat. 
- Added README.

### CODE REVIEW

File: jobs/dtd-metrics/src/index.ts [hunk 1]
```diff
@@ -12,81 +11,60 @@ import {
...
+})
 
+try {
   const githubClient = new GithubClient(env.GITHUB_ACCESS_TOKEN)
   const awsS3BucketClient = new AwsS3BucketClient(env.STORAGE_BUCKET)
 
-  log('Retrieving metrics...')
+  log.info('Initializing global store...')
+  const globalStore = await GlobalStoreService.init(readModel, { cache: env.CACHE_GLOBAL_STORE })
+  log.info('Global store initialized!\n')
```
Comment (galales) [inline jobs/dtd-metrics/src/index.ts]: new lines in logs create multiple entries on cloudwatch with empty data, and should be avoided

File: jobs/dtd-metrics/src/services/metrics-producer.service.ts [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+   */
+  public async produceOutput({
+    produceJSON,
+    filter,
+  }: ProduceOutputOptions): Promise<MetricsOutput | Partial<MetricsOutput>> {
+    const metricsOutput: Record<string, unknown> = {}
+
+    for (const metricObj of this.metrics) {
+      // If a filter is specified, skip the metric if its name does not include the filter
+      if (filter && !metricObj.name.includes(filter)) continue
```
Comment (galales) [inline jobs/dtd-metrics/src/services/metrics-producer.service.ts:53]: not sure about the purpose of this filter, is it only for development?
if not, shouldn't be better if we entirely avoid the calculation if a metric is not required?

File: jobs/dtd-metrics/src/services/metrics-producer.service.ts [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+      metricsOutput[metricObj.name] = await metricObj.factoryFn(this.readModel, this.globalStore)
+    }
+
+    const output = filter ? MetricsOutput.partial().parse(metricsOutput) : MetricsOutput.parse(metricsOutput)
+
+    if (produceJSON) {
+      writeFileSync('dtd-metrics.json', JSON.stringify(output, null, 2))
```
Comment (galales) [inline jobs/dtd-metrics/src/services/metrics-producer.service.ts]: may I suggest to move this write [here](https://github.com/pagopa/interop-be-reports/pull/52/files?diff=split&w=1#diff-4a6f3d35f67b85ad66ce4d63bfa23ec5dba19f22952a2d4e709ddbcf233b0573R60)?
Having all outputs operations in one place avoids the need of looking through the code for each write, here is a bit hidden

**Review by galales — COMMENTED** [review]
for future changes, please create a PR for each feature, especially when applying format or major refactors on the entire code base, it's quite difficult to see the main changes with all the noise 😅

File: jobs/dtd-metrics/src/services/metrics-producer.service.ts (hunk 2 above)
Comment (Carminepo2) [reply jobs/dtd-metrics/src/services/metrics-producer.service.ts:53]: It is only for development use. It helps a lot when I need to concentrate on a specific metric. Do you think I should make this clear?

Comment (galales) [reply jobs/dtd-metrics/src/services/metrics-producer.service.ts:53]: if it's just for development it's fine, not my intention to overcomplicate the logic if not required 😁

**Review by galales — APPROVED** [review]

**Review by beetlecrunch — APPROVED** [review]

**Pull request merged**